src/main/java/core/basesyntax/service/FileServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class FileServiceImpl implements FileService {
+    private static final String HEAD_OF_REPORT = "fruit,quantity";
+
+    @Override
+    public List<String> readFromFile(String fileName) {
+        List<String> lineArray = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                lineArray.add(line);
+            }
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File not found" + fileName, e);
+        } catch (IOException e) {
+            throw new RuntimeException("Cant read file" + fileName, e);
+        }
+
+        return lineArray;
+    }
+
+    @Override
+    public boolean createReport(Map<String, Integer> map, String fileName) {
```

Comment:
this method contradicts single-responsibility principle of class
if it's FileService and it has public List<String> readFromFile(String fileName)
then second method is public void writeToFile(String report, String fileName)

Please rename and extract logic for report creation to separate service